python/plugins/processing/algs/qgis/PointDistance.py

```diff
@@ -116,11 +116,17 @@ def processAlgorithm(self, parameters, context, feedback):
         if source is None:
             raise QgsProcessingException(self.invalidSourceError(parameters, self.INPUT))
 
+        if QgsWkbTypes.isMultiType(source.wkbType()):
+            raise QgsProcessingException(self.tr('Input point layer is a MultiPoint layer - first convert to single points before using this algorithm'))
```

Comment:
Consider adding full stop at the end?
I did not check but should we have this for shortest path algs too?

Reply:
There's probably *many* algorithms which could have this same issue -- https://github.com/qgis/QGIS/pull/8673 at least prevents them from giving misleading results. We'd need to identify these others and give friendly messages like this one too.